float64 coordinate_x
float64 coordinate_y
float64 angle